File: 02_recursion_basics/02_07_fibonacci/Fibonacci.java
public class Fibonacci {
    public static void main(String[] args) {
        int num = 7;
        System.out.println(String.format("Fibonacci for %d: %d", num, fibonacci(num)));
    }

    private static int fibonacci(int n) {
        if (n <= 1) {
            return n;
        }
        int a = 0;
        int b = 1;

        int temp = n;

        while (temp-- > 1) {
            int sum = a + b;
            a = b;
            b = sum;
        }

        return b;
    }
}

File: 02_recursion_basics/02_07_fibonacci/fibonacci.cpp
#include <bits/stdc++.h>

using namespace std;

int fibonacci(int n) {
    if (n == 0) {
        return 0;
    }

    if (n == 1) {
        return 1;
    }

    return fibonacci(n - 1) + fibonacci(n - 2);
}

int main() {
    int num = 7;
    cout << "Fibonacci for: " << num << ": " << fibonacci(num) << endl;
}